File: simple.python/344.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        print()
        if s == None:
            return ""
        a = s[::-1]
        s = a
        print(s)

    # 执行用时：52 ms, 在所有 Python3 提交中击败了22.66%的用户
    # 内存消耗：19.6 MB, 在所有 Python3 提交中击败了36.98%的用户
    # 通过测试用例：477 / 477
    def reverseString2(self, s: List[str]) -> None:
        end = len(s) // 2
        print(end)
        for i in range(0, end):
            tmp = s[i]
            s[i] = s[len(s) - i - 1]
            s[len(s) - i - 1] = tmp
        print(s)


def main():
    a_solution = Solution()
    a_solution.reverseString2(["h","e","l","l","o"])
    pass

main()

File: simple.golang/344.go
package main

// 执行用时：20 ms, 在所有 Go 提交中击败了99.92%的用户
// 内存消耗：6.4 MB, 在所有 Go 提交中击败了71.41%的用户
// 通过测试用例：477 / 477
func reverseString(s []byte) {
	var end int
	end = len(s) / 2
	var i int
	var tmp byte
	for i = 0; i < end; i++ {
		tmp = s[i]
		s[i] = s[len(s)-i-1]
		s[len(s)-i-1] = tmp
	}
}
